pkg/executables/flux.go

```diff
@@ -63,17 +67,61 @@ func (f *Flux) BootstrapToolkitsComponentsGithub(ctx context.Context, cluster *t
 
 	_, err = f.ExecuteWithEnv(ctx, env, params...)
 	if err != nil {
-		return fmt.Errorf("executing flux bootstrap: %v", err)
+		return fmt.Errorf("executing flux bootstrap github: %v", err)
 	}
 
 	return err
 }
 
-// BootstrapToolkitsComponentsGit creates the Git repository if it does not exist, and commits the toolkit
-// components manifests to the main branch. Then it configures the target cluster to synchronize with the repository.
-// If the toolkit components are present on the cluster, the bootstrap command will perform an upgrade if needed.
+// BootstrapToolkitsComponentsGit commits the toolkit components manifests to the branch of a Git repository.
+// It then configures the target cluster to synchronize with the repository. If the toolkit components are present on the cluster, the
+// bootstrap command will perform an upgrade if needed.
 func (f *Flux) BootstrapToolkitsComponentsGit(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error {
-	return nil
+	c := fluxConfig.Spec
```

Comment:
This is almost like copy paste from `BootstrapToolkitsComponentsGithub`
Can we have a common private method that both of them use?